src/operator/convolution.js

```diff
@@ -0,0 +1,62 @@
+'use strict'
+
+import edgeHandlingMethod from './edgeHandlingMethod'
+
+//convolution using a square kernel
+export default function convolution(newImage, kernel, edgeHandling){
+    let kernelWidth = Math.sqrt(kernel.length);
+    let k = Math.floor(kernelWidth/2);
+    let div = 0;
+    edgeHandling = edgeHandling || 'copy';
+
+    for(let i = 0; i < kernel.length; i++){div += kernel[i];}
+
+    for(let x = 0; x < this.width ; x++){
+        for(let y = 0; y < this.height ; y++){
+            let sum = 0;
+            for(let i = -k; i <= k; i++){
+                for(let j = -k; j <= k; j++){
+                    let val = edgeHandling.toLowerCase() == 'mirror'
```

Comment:
please do not use ternary operator for more than one condition. It is unclear what this statement is doing